airflow/providers/cncf/kubernetes/operators/resource.py

```diff
@@ -61,20 +61,22 @@ def __init__(
         yaml_conf: str,
         namespace: str | None = None,
         kubernetes_conn_id: str | None = KubernetesHook.default_conn_name,
+        config_file: str | None = None,
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
         self._namespace = namespace
         self.kubernetes_conn_id = kubernetes_conn_id
         self.yaml_conf = yaml_conf
+        self.config_file = config_file
 
     @cached_property
     def client(self) -> ApiClient:
         return self.hook.api_client
 
     @cached_property
     def hook(self) -> KubernetesHook:
-        hook = KubernetesHook(conn_id=self.kubernetes_conn_id)
+        hook = KubernetesHook(conn_id=self.kubernetes_conn_id, config_file=self.config_file)
```

Comment:
Dont you think we should have a None check before using the config_file here? If not passed, it will go as None to the hook. 

Also can you add some test coverage for this?

Reply:
I updated the existing tests for the modified operators in the last commit